src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -1133,6 +1134,120 @@ export class ShowWorkspaceRecommendedExtensionsAction extends Action {
+		@IExtensionsWorkbenchService private extensionsWorkbenchService: IExtensionsWorkbenchService,
+		@IExtensionTipsService private extensionTipsService: IExtensionTipsService,
+		@IMessageService private messageService: IMessageService
+	) {
+		super(id, label, 'extension-action');
+		this.extensionsWorkbenchService.onChange(() => this.updateToInstall(), this, this.disposables);
+		this.contextService.onDidChangeWorkbenchState(() => this.updateRecommendations(), this, this.disposables);
+		this.updatePromise = this.updateRecommendations();
+	}
+
+	private updateRecommendations(): TPromise<any> {
+		this.enabled = this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY;
+		if (this.enabled) {
+			return this.extensionTipsService.getWorkspaceRecommendations().then(names => {
+				this.recommendations = names;
+				this.updateToInstall();
+			});
+		}
+		return TPromise.as(null);
+	}
+
+	private updateToInstall(): void {
+		const installed = this.extensionsWorkbenchService.local.map(x => x.id.toLowerCase());
+		this.toInstall = this.recommendations.filter(x => installed.indexOf(x.toLowerCase()) === -1);
+		this.enabled = this.toInstall.length > 0;
+	}
+
+	run(): TPromise<any> {
+		this.viewletService.openViewlet(VIEWLET_ID, true)
+			.then(viewlet => viewlet as IExtensionsViewlet)
+			.then(viewlet => {
+				viewlet.search('@recommended:workspace ');
+				viewlet.focus();
+			});
+
+		return this.updatePromise.then(() => {
+			if (!this.toInstall.length) {
+				this.messageService.show(Severity.Info, localize('allExtensionsInstalled', "All extensions recommended for this workspace have already been installed"));
+				return TPromise.as(null);
+			}
+			return this.extensionsWorkbenchService.queryGallery({ names: this.toInstall, source: 'install-all-workspace-recommendations' }).then(pager => {
+				return TPromise.join(pager.firstPage.map(e => this.extensionsWorkbenchService.install(e)));
+			});
+		});
+	}
+
+	protected isEnabled(): boolean {
+		return this.enabled;
+	}
+
+	dispose(): void {
+		this.disposables = dispose(this.disposables);
+		super.dispose();
+	}
+}
+
+export class InstallRecommendedExtensionAction extends Action {
+
+	static ID = 'workbench.extensions.action.installRecommendedExtension';
+	static LABEL = localize('installRecommendedExtension', "Install Recommended Extension");
+
+	private extensionId: string;
+
+	constructor(
+		extensionId: string,
+		@IWorkspaceContextService private contextService: IWorkspaceContextService,
+		@IViewletService private viewletService: IViewletService,
+		@IExtensionsWorkbenchService private extensionsWorkbenchService: IExtensionsWorkbenchService,
+		@IExtensionTipsService private extensionTipsService: IExtensionTipsService
+	) {
+		super(InstallRecommendedExtensionAction.ID, InstallRecommendedExtensionAction.LABEL, null);
+		this.extensionId = extensionId;
+	}
+
+	run(): TPromise<any> {
+		this.viewletService.openViewlet(VIEWLET_ID, true)
```

Comment:
Not sure what is the intention here. To show recommended extensions before install or after install. Data will be stale if you do it before and does not makes sense if it is shown after. Since, this is an install action, how about showing Installed extensions after installing?

Reply:
Stale is ok.
If the installation was fast, then the view would show the current recommendation with the "Reload" button.
If the installation was slow, then the view would show the current recommendation with the "Installing.." button.

The intention is also to show rest of the recommendations when installing the current recommendation.